### PR TITLE
Add t8n features required by exe-spec-tests

### PR DESCRIPTION
**Pull request by rodiazet**

### CODE REVIEW

File: test/t8n/t8n.cpp [hunk 1]
```diff
@@ -137,6 +138,7 @@ int main(int argc, const char* argv[])
 
                         j_receipt["transactionHash"] = hex0x(computed_tx_hash);
                         j_receipt["gasUsed"] = hex0x(static_cast<uint64_t>(receipt.gas_used));
+                        blockGasUsed += receipt.gas_used;
```
Comment (chfast) [inline test/t8n/t8n.cpp:141]: This is also `"cumulativeGasUsed"` so I propose to name the variable like this and assign correct value to the receipt. The block gas used it the value of the last receipt.

File: test/t8n/t8n.cpp [hunk 2]
```diff
@@ -82,6 +82,7 @@ int main(int argc, const char* argv[])
 
         std::vector<state::Transaction> transactions;
         std::vector<state::TransactionReceipt> receipts;
+        int64_t blockGasUsed = 0;
```
Comment (chfast) [inline test/t8n/t8n.cpp:85]: Put it before vectors.

File: test/t8n/t8n.cpp [hunk 3]
```diff
@@ -80,6 +83,7 @@ int main(int argc, const char* argv[])
         j_result["currentDifficulty"] = "0x20000";
         j_result["currentBaseFee"] = hex0x(block.base_fee);
 
+        int64_t cumulativeGasUsed = 0;
```
Comment (chfast) [inline test/t8n/t8n.cpp:86]:
```suggestion
        int64_t cumulative_gas_used = 0;
```

**Review by chfast — COMMENTED** [review]
Looks good, just let's merge #603 first.

File: test/t8n/t8n.cpp (hunk 3 above)
Comment (rodiazet) [reply test/t8n/t8n.cpp:86]: Shit. Sorry for this. Too fast as always.

**Review by chfast — APPROVED** [review]

**Pull request merged**